terraform-provider-constellation/internal/provider/attestation_data_source.go

```diff
@@ -191,27 +126,23 @@ func (d *AttestationDataSource) Read(ctx context.Context, req datasource.ReadReq
 		resp.Diagnostics.AddAttributeError(
 			path.Root("attestation_variant"),
 			"Invalid Attestation Variant",
-			fmt.Sprintf("Invalid attestation variant: %s", data.CSP.ValueString()),
+			fmt.Sprintf("Invalid attestation variant: %s", data.AttestationVariant.ValueString()),
 		)
 		return
 	}
+
+	snpVersions := attestationconfigapi.SEVSNPVersionAPI{}
 	if attestationVariant.Equal(variant.AzureSEVSNP{}) || attestationVariant.Equal(variant.AWSSEVSNP{}) {
-		snpVersions, err := d.fetcher.FetchSEVSNPVersionLatest(ctx, attestationVariant)
+		snpVersions, err = d.fetcher.FetchSEVSNPVersionLatest(ctx, attestationVariant)
 		if err != nil {
 			resp.Diagnostics.AddError("Fetching SNP Version numbers", err.Error())
 			return
 		}
-		tfSnpAttestation, err := convertSNPAttestationTfStateCompatible(attestationVariant, snpVersions)
-		if err != nil {
-			resp.Diagnostics.AddError("Converting SNP attestation", err.Error())
-		}
-		diags := resp.State.SetAttribute(ctx, path.Root("attestation"), tfSnpAttestation)
-		resp.Diagnostics.Append(diags...)
-		if resp.Diagnostics.HasError() {
-			return
-		}
 	}
-
+	tfSnpAttestation, err := convertToTfAttestation(attestationVariant, snpVersions)
```

Comment:
Can you rename all occurrences of `tfSnpAttestation` to simply `tfAttestation` (or similar)?
If I understand this code correctly, this struct is not exclusively used for just SNP based attestation, so it shouldn't be named for that.